### DISCUSSION TITLE
Adding constraints

### DISCUSSION THREAD

**Discussion by fsikora** [ds4dm/ml4co-competition#4] [Q&A] [answered]
Hi,

We're wondering if you allow the adding of constraints into the model
during the solving process, either in task 1 or 2.

The first item on the "cheating prevention" paragraph is unclear to us
about this specific question.

best

**Comment by gasse** [comment, marked as answer]
Hi @fsikora ,

Thank you for this question !

We don't allow any interaction with the solver, besides the extraction of information (through an observation function) at every time step, and the choice of the action sent to the environment (either a partial solution, a branching variable, or a set of parameters). This restriction is imposed so that each task is a well-defined control problem and remains conceptually simple.

Best,
Maxime
